src/static_handler.rs

```diff
@@ -168,25 +168,43 @@ impl Cache {
         };
 
         let if_modified_since = match req.headers.get::<IfModifiedSince>().cloned() {
-            None => return Ok(self.response_with_cache(path, last_modified_time)),
+            None => return Ok(self.response_with_cache(req, path, last_modified_time)),
             Some(IfModifiedSince(HttpDate(time))) => time.to_timespec(),
         };
 
         if last_modified_time <= if_modified_since {
             Ok(Response::with(status::NotModified))
         } else {
-            Ok(self.response_with_cache(path, last_modified_time))
+            Ok(self.response_with_cache(req, path, last_modified_time))
         }
     }
 
-    fn response_with_cache<P: AsRef<Path>>(&self, path: P, modified: Timespec) -> Response {
+    fn response_with_cache<P: AsRef<Path>>(&self, req: &mut Request, path: P, modified: Timespec) -> Response {
         use iron::headers::{CacheControl, LastModified, CacheDirective, HttpDate};
+        use iron::headers::{ContentLength, ContentType};
+        use iron::method::Method;
+        use iron::mime::{Mime, TopLevel, SubLevel};
+        use iron::modifiers::Header;
 
-        let mut response = Response::with((status::Ok, path.as_ref()));
+        let mut response: Response;
         let seconds = self.duration.as_secs() as u32;
         let cache = vec![CacheDirective::Public, CacheDirective::MaxAge(seconds)];
+        let metadata = fs::metadata(path.as_ref()).unwrap();
```

Comment:
In case of lacking permissions I would return a 500 Internal Server Error. However, in such a case it would be useful to return an `Err(IronError)` instead of `Ok(Response)` which changes the type signature of this function.